KEYCLOAK-10313 Add PKCE support to KeycloakInstalled Adpater

thomasdarimont opened the pull request:
This adds PKCE support for Desktop Apps as
a followup to KEYCLOAK-1033 #6047.

thomasdarimont committed:
KEYCLOAK-10313 Add PKCE support to KeycloakInstalled Adpater

This adds PKCE support for Desktop Apps as
a followup to KEYCLOAK-1033 #6047.

mposolda commented on adapters/oidc/installed/src/main/java/org/keycloak/adapters/installed/KeycloakInstalled.java:
```diff
@@ -161,17 +165,9 @@ public void loginDesktop() throws IOException, VerificationException, OAuthError
 
         String redirectUri = "http://localhost:" + callback.server.getLocalPort();
         String state = UUID.randomUUID().toString();
+        Pkce pkce = generatePkce();
```
It seems this always enables pkce without possibility to disable it. I think it can be enabled just if "deployment.isPkce()" is true?

Is it also possible to update docs for KeycloakInstalled adapter and add the info about pkce support?

thomasdarimont replied:
Thanks for the hint! I just added a check for `enable-pkce`

mposolda reviewed (commented):
@thomasdarimont Thanks! Left just one comment.

mposolda commented:
@thomasdarimont Any update?

thomasdarimont committed:
KEYCLOAK-10313 Only use PKCE if enable-pkce is configured for KeycloakInstalled adapter

Users who want to use PKCE support with the KeycloakInstalled adapter need to set the property
``"enable-pkce": true` in the adapter configuration / `keycloak.json`.

thomasdarimont commented:
@mposolda  Added PR for Documentation update
https://github.com/keycloak/keycloak-documentation/pull/687

mposolda reviewed (approved):
@thomasdarimont Thanks!
